components/TimeSeries.vue

```diff
@@ -0,0 +1,353 @@
+
+import { highchartsColors, plotBandsColor, plotLinesColor } from "./utils/charts";
+import type { DisplayInfo } from "~/types/apiResponseTypes";
+
+const props = defineProps<{
+  seriesId: string
+  index: number
+  openedAccordions: string[]
+  hideTooltips: boolean
+}>();
+
+const emit = defineEmits(["toggleOpen"]);
+accessibilityInitialize(Highcharts);
+exportingInitialize(Highcharts);
+exportDataInitialize(Highcharts);
+offlineExportingInitialize(Highcharts);
+
+const appStore = useAppStore();
+
+let chart: Highcharts.Chart;
+const chartBackgroundColor = "transparent";
+const chartBackgroundColorOnExporting = "white";
+const hideResetZoomButtonClassName = "hide-reset-zoom-button";
+const hideTooltipsClassName = "hide-tooltips";
+const accordionBodyYPadding = 8;
+const accordionStyle = {
+  "--cui-accordion-btn-focus-box-shadow": "none",
+  "--cui-accordion-bg": "rgba(255, 255, 255, 0.7)",
+};
+// We need each chart to have a higher z-index than the next one so that the exporting context menu is always on top and clickable.
+// Also, they should be at least 3 so that they are above .accordion-button:focus
+const zIndex = (Object.keys(appStore.timeSeriesData!).length - props.index) + 3;
+const yUnits = props.seriesId === "dead" ? "deaths" : "cases"; // TODO: Make this depend on a 'units' property in metadata. https://mrc-ide.myjetbrains.com/youtrack/issue/JIDEA-117/
+const minAccordionHeight = 150;
+const minTotalAccordionHeight = 500;
+
+const isOpen = computed(() => props.openedAccordions.includes(props.seriesId));
+const maxTotalAccordionHeight = computed(() => {
+  if (appStore.timeSeriesData) { // Allow at least minAccordionHeight for each accordion
+    return Math.max(minTotalAccordionHeight, (Object.keys(appStore.timeSeriesData!).length * minAccordionHeight));
+  } else {
+    return minTotalAccordionHeight;
+  }
+});
+// Share available height equally between open accordions
+const containerHeightPx = computed(() => maxTotalAccordionHeight.value / props.openedAccordions.length);
+const containerId = computed(() => `${props.seriesId}-container`);
+// Assign an x-position to y-values. Nth value corresponds to "N+1th day" of simulation.
+const data = computed(() => {
+  return appStore.timeSeriesData![props.seriesId].map((value, index) => [index + 1, value]);
+});
+const seriesMetadata = computed((): DisplayInfo | undefined => {
+  return appStore.metadata?.results?.time_series.find(({ id }) => id === props.seriesId);
+});
+const usePlotLines = props.seriesId === "hospitalised"; // https://mrc-ide.myjetbrains.com/youtrack/issue/JIDEA-118/
+const usePlotBands = props.seriesId === "hospitalised" || props.seriesId === "infect"; // https://mrc-ide.myjetbrains.com/youtrack/issue/JIDEA-118/
+// On zooming, the y-axis automatically rescales to the data (ignoring the plotLines). We want the plotLines
+// to remain visible, so we limit the y-axis' ability to rescale, by defining a minimum range.
```

Comment:
```suggestion
// On zooming, the y-axis automatically rescales to the data (ignoring the plotLines). We want the plotLines
// to remain visible, so we limit the y-axis' ability to rescale, by defining a minimum range. This way the plotLines remain visible even when the maximum data value is less than the maximum plotline value.
```
Took me a couple of goes to understand this!